use md-to-pdf@3.0.0

simonhaenisch opened the pull request:
Hi, I'm the creator of `md-to-pdf` and I've just released version 3 last weekend, which brings a couple improvements to the API. One of them is that it's now possible to get the pdf content as a `Buffer`, rather than having it written to disk as a file automatically. This saves you a lot of steps with creating temporary files and having to clean them up.

This project is probably the biggest user of `md-to-pdf` at the moment, and I've been using dillinger.io for quite a while, so I thought I'd help you with integrating the upgrade into your server.

I wasn't able to run the tests locally, but previewing and exporting as PDF works fine on localhost (using `npm start`).

joemccann commented:
This is so awesome thanks!

joemccann commented:
The key will be to see if it works in a containerized environment? Locally has been fine because Chromium is installed.

joemccann commented:
Still the same issue :(

simonhaenisch commented:
I've been able to get it to run in CircleCI in the past (now using Github Actions), but I've had to use a different container, `circleci/node:8-browsers` instead of `circleci/node-8`, which has some stuff for Firefox and Chrome/Chromium pre-configured. With that I was able to run md-to-pdf in a containerized environment.

joemccann commented:
Yeah, unfortunately, Zeit doesn't give me that option IIRC.  It's such a
bummer as this is the best feature of Dillinger!

On Mon, Mar 2, 2020 at 4:37 PM Simon Hänisch <notifications@github.com>
wrote:

> I've been able to get it to run in CircleCI in the past (now using Github
> Actions), but I've had to use a different container,
> circleci/node:8-browsers instead of circleci/node-8, which has some stuff
> for Firefox and Chrome/Chromium pre-configured. With that I was able to run
> md-to-pdf in a containerized environment.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/joemccann/dillinger/pull/748?email_source=notifications&email_token=AAALY2RU2I47N6IQA4SAOTDRFRGNHA5CNFSM4K7SH4ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENRTU2Y#issuecomment-593705579>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALY2STZKL4MSHGMCMLIXTRFRGNHANCNFSM4K7SH4ZA>
> .
>

simonhaenisch commented:
Hm just looking at https://zeit.co/blog/serverless-chrome (which is a response to your issue https://github.com/zeit/now/issues/1535)... I think it would be possible!

You can pass custom launch options into `md-to-pdf`. I'll give it a try and let you know!

simonhaenisch commented:
Hm it should be possible with Now v2 to deploy an api route with [serverful behavior](https://zeit.co/docs/runtimes#advanced-usage/advanced-node-js-usage/legacy-serverful-behavior) (I moved `app.js` to `api/app.js`, added a rewrite from `/(.*)` to `/api/app` and removed everything else from now.json), and I'm able to deploy it but it doesn't run because `config.js` always throws an error (something about path.join expected string but got undefined) 🤷🏻‍♂️

If you want to give it a try yourself, I think you'd just need to install the `chrome-aws-lambda` package, then in `plugins/core/server.js` use it like

```js
const chrome = require('chrome-aws-lambda')

pdf = await mdToPdf({ content: md }, {
  launch_options: {
    args: chrome.args,
    executablePath: await chrome.executablePath,
    headless: chrome.headless,
  }
})
```

joemccann commented:
awesome will give this a go!

On Tue, Mar 3, 2020 at 1:08 AM Simon Hänisch <notifications@github.com>
wrote:

> Hm it should be possible with Now v2 to deploy an api route with serverful
> behavior
> <https://zeit.co/docs/runtimes#advanced-usage/advanced-node-js-usage/legacy-serverful-behavior>
> (I moved app.js to api/app.js, added a rewrite from /(.*) to /api/app and
> removed everything else from now.json), and I'm able to deploy it but it
> doesn't run because config.js always throws an error (something about
> path.join expected string but got undefined) 🤷🏻‍♂️
>
> If you want to give it a try yourself, I think you'd just need to install
> the chrome-aws-lambda package, then in plugins/core/server.js use it like
>
> const chrome = require('chrome-aws-lambda')
>
>
>
> pdf = await mdToPdf({ content: md }, {
>
>   launch_options: {
>
>     args: chrome.args,
>
>     executablePath: await chrome.executablePath,
>
>     headless: chrome.headless,
>
>   }
>
> })
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/joemccann/dillinger/pull/748?email_source=notifications&email_token=AAALY2V5CXI7HUF237WLBPLRFTCH5A5CNFSM4K7SH4ZKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENSVHOI#issuecomment-593843129>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALY2R3G4YNTS45LK3OFATRFTCH5ANCNFSM4K7SH4ZA>
> .
>
